k8s/envoy-opa/README.md

```diff
@@ -0,0 +1,346 @@
+
+```console
+   - name: envoy.ext_authz
+     typed_config:
+       "@type": type.googleapis.com/envoy.config.filter.http.ext_authz.v2.ExtAuthz
+       with_request_body:
+         max_request_bytes: 8192
+         allow_partial_message: true
+       failure_mode_allow: false
+       grpc_service:
+         google_grpc:
+           target_uri: 127.0.0.1:8182
+           stat_prefix: ext_authz
+         timeout: 0.5s
+```
+
+Note that `target_uri` is configured to talk to the `OPA` service defined in a [previous step](#opa-configuration).
+If you are curious, the complete configuration file is located at [k8s/backend/config/envoy.yaml](k8s/backend/config/envoy.yaml).
+
+## Apply the New Resource
+
+In order for the new configuration to take effect, the configmap for the OPA configuration needs to be applied and Envoy configuration needs to be updated. Ensure that the current working directory is `.../spire-tutorials/k8s/envoy-opa` and apply the new configurations using:
+
+```console
+   $ kubectl apply -k k8s/.
+   
+   configmap/backend-envoy configured
+   configmap/backend-opa-policy-config configured
+   deployment.apps/backend configured
+```
+
+Next, the `backend` pod needs to be restarted to pick up the new configurations:
+
+```console
+   $ kubectl scale deployment backend --replicas=0
+   $ kubectl scale deployment backend --replicas=1
+```
+
+
+# Part 2: Test connections
+
+Now that services are deployed and also registered in SPIRE, let's test the authorization that we've configured.
+
+## Testing Valid Requests
+
+The first test will demonstrate how a request that satisfies the policy allow for the display of associated data. To run this test, we need to find the IP address and port that make up the URL to use for accessing the data.
+
+```console
+   $ kubectl get services
+
+   NAME            TYPE           CLUSTER-IP    EXTERNAL-IP      PORT(S)          AGE
+   backend-envoy   ClusterIP      None          <none>           9001/TCP         6m53s
+   frontend        LoadBalancer   10.8.14.117   35.222.164.221   3000:32586/TCP   6m52s
+   frontend-2      LoadBalancer   10.8.7.57     35.222.190.182   3002:32056/TCP   6m53s
+   kubernetes      ClusterIP      10.8.0.1      <none>           443/TCP          59m
+```
+
+The `frontend` service will be available at the `EXTERNAL-IP` value and port `3000`, which was configured for our container. In the sample output shown above, the URL to navigate is `http://35.222.164.221:3000`. Open your browser and navigate to the IP address shown for `frontend` in your environment, adding the port `:3000`. Once the page is loaded, you'll see the account details for user _Jacob Marley_. 
+
+![Frontend][frontend-view]
+
+[frontend-view]: images/frontend_view.png "Frontend view"
+
+Let's take a look at the OPA Agent logs to see what is happening behind the scenes. Use the following Bash script to get the logs for the OPA instance running next to the `backend` service on a pretty print format: 
```

Comment:
Hmm, the world is a big place. I'm not sure it's safe to assume all people who run this demo will have jq installed. I kind of do think jq should be listed as a requirement. But if you disagree, at least let's make jq link to its web page: https://stedolan.github.io/jq/ . (We do assume users have git installed and don't mention it...) It's ok if it can't be both monospace and a link.

Reply:
Ok, sounds good to me. I'll add the link to the reference.
Thanks.